iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/ActivePowerControlImpl.java

```diff
@@ -26,22 +27,38 @@ public class ActivePowerControlImpl<T extends Injection<T>> extends AbstractMult
     private final TDoubleArrayList droop;
     private final TDoubleArrayList participationFactor;
 
+    private final TDoubleArrayList minPOverride;
+    private final TDoubleArrayList maxPOverride;
+
     private final List<TDoubleArrayList> allTDoubleArrayLists;
 
     public ActivePowerControlImpl(T component,
                                   boolean participate,
                                   double droop,
                                   double participationFactor) {
+        this(component, participate, droop, participationFactor, Double.NaN, Double.NaN);
+    }
+
+    public ActivePowerControlImpl(T component,
+                                  boolean participate,
+                                  double droop,
+                                  double participationFactor,
+                                  double minPOverride,
+                                  double maxPOverride) {
         super(component);
         int variantArraySize = getVariantManagerHolder().getVariantManager().getVariantArraySize();
         this.participate = new TBooleanArrayList(variantArraySize);
         this.droop = new TDoubleArrayList(variantArraySize);
         this.participationFactor = new TDoubleArrayList(variantArraySize);
-        this.allTDoubleArrayLists = List.of(this.droop, this.participationFactor);
+        this.minPOverride = new TDoubleArrayList(variantArraySize);
```

Comment:
Why do you need a variant dependency?

Reply:
@annetill 
Well... Why not ? You may imagine scenarios where you play with the compensation margin.

But the reality is that I discover now the concept of variants possible for some fields and not others. A strange concept.
Especially if it can vary in the fields of a same extension. Is this concept driven by real needs or by the memory heavy implementation of variants ?

Anyway, let me know what you want. My intention was juts to be homogenous with the existing local code and I had not strong reflections on variants.

Reply:
Because it is not with minP and maxP as attributes of a generator, so it is not homogeneous.

Reply:
Let's discuss. I do not like the homogenous argument.
First because, making decision based on consistency leads nowhere.
Second because we have the choice between two inconsisencies. In the extension or vs the pMax/pMin of the generator.
I'd prefer to discuss based on need and tradeoff to make because of the high cost of current variant implementation.
Let's discuss on Monday.